Sparse_and_dense.ipynb

```diff
@@ -0,0 +1,320 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from scipy import sparse\n",
+    "import sys\n",
+    "import os\n",
+    "import time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 89,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.random.seed(0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Sparse .npz (compressed) takes the least disk space regardless of density\n",
+    "Uncompressed sparse matrices take more disk space when the matrices are very dense"
```

Comment:
I look to source.

When scipy saves sparse matrices, [it uses](https://github.com/scipy/scipy/blob/v1.0.0/scipy/sparse/_matrix_io.py#L76-L79) `numpy.savez` or `numpy.savez_compressed`. It's saving the indices and offsets (arrays) just like yi was in HDF5. This now makes sense why they use .npz extension.

Then we take a look at numpy's `savez` and `savez_compressed` [source. It's using ZIP.](https://github.com/numpy/numpy/blob/v1.14.0/numpy/lib/npyio.py#L689-L698)